packages/lib/src/plugin.ts

```diff
@@ -25,7 +25,35 @@ export type BasePlugin =
 ;
 
 // TODO Add proper typing for plugins
-export type PluginInfo = any;
+/* Used to define the info.ts of plugins, plugin_loader transform it into PluginInfo */
+export type PluginInfoTs = {
+	name: string;
+	title: string;
+	description?: string;
+
+	instanceEntrypoint?: string;
+	InstanceConfigGroup?: typeof PluginConfigGroup;
+
+	controllerEntrypoint?: string;
+	ControllerConfigGroup?: typeof PluginConfigGroup;
+
+	controlEntrypoint?: string;
+	ControlConfigGroup?: typeof PluginConfigGroup;
+
+	messages?: any[];
+
+	webEntrypoint?: string;
+	routes?: string[];
+}
+export type PluginInfo = PluginInfoTs & {
+	requirePath: string;
+	version: string;
+	manifest: any;
+	container: any;
+	package: any;
+	enabled?: boolean;
+};
```

Comment:
Is `PluginInfo` and ~~PluginInfoTs~~ `PluginInfoTsFileExport` better ?

Reply:
No. One is a plugin declaration and the other is the public declaration plus an amalgamation of the Node.js enviroment information about a plugin and the Webpack environment information about a plugin. These should not both be called "info". Also "Ts" does not belong in the name because it has nothing to do with TypeScript, a pure JavaScript module would export the same object conforming to the same type.